File: 백준/Bronze/2231. 분해합/분해합.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

       int num = sc.nextInt();

        for (int i = 1; i < num; i++) {

            int sum = i;
            int tmp = i;

            while(tmp != 0) {
                sum += tmp % 10;
                tmp /= 10;
            }

            if (sum == num) {
                System.out.println(i);
                return;
            }
        }

        System.out.println(0);
    }
}

File: 백준/Bronze/2231. 분해합/분해합.cc
#include <iostream>
using namespace std;

int main() {

	int num;
	cin >> num;

	for (int i = 1; i < num; i++)
	{
		int sum = i;
		int tmp = i;

		while (true)
		{
			sum += tmp % 10;
			tmp /= 10;

			if (tmp == 0)
			{
				break;
			}
		}

		if (sum == num)
		{
			cout << i << endl;
			return 0;
		}
	}

	cout << 0 << endl;
}